geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -1308,16 +1207,9 @@ public void checkIdleExpiration() {
       if (expireCount > 0) {
         getPoolStats().incIdleExpire(expireCount);
         getPoolStats().incPoolConnections(-expireCount);
-        // do this outside the above sync
-        lock.lock();
-        try {
-          connectionCount -= expireCount;
-          freeConnection.signalAll();
-          if (connectionCount < minConnections) {
-            startBackgroundPrefill();
-          }
-        } finally {
-          lock.unlock();
+        connectionCount.addAndGet(-expireCount);
+        if (connectionCount.get() < minConnections) {
```

Comment:
instead of calling the atomic's get a second time (addAndGet called it) just use the result of addAndGet for this comparison.

Reply:
Yup, keen eye!